library/src/main/java/com/novoda/downloadmanager/CallbackThrottle.java

```diff
@@ -1,10 +1,27 @@
 package com.novoda.downloadmanager;
 
+/**
+ * For defining the speed at which {@link DownloadBatchStatus} notifications are emitted.
+ */
 public interface CallbackThrottle {
 
+    /**
+     * The callback to pass each emission of {@link DownloadBatchStatus} to.
+     *
+     * @param callback to notify.
+     */
     void setCallback(DownloadBatchStatusCallback callback);
 
+    /**
+     * Called internally by the download manager each time a {@link DownloadBatchStatus} changes.
+     * Clients should control the throttling of this status.
+     *
+     * @param downloadBatchStatus to throttle.
+     */
     void update(DownloadBatchStatus downloadBatchStatus);
 
+    /**
+     * Prevent additional {@link DownloadBatchStatus} from being emitted.
```

Comment:
I don't follow. This method prevents additional statuses from being emitted. Either with a guard statement or by removing a listener. It doesn't notify any callback to stop doing something.

Reply:
What I meant is that given how it's phrased like now, as a client I'm implementing this interface with my concrete implementation. This method is gonna then be called to tell my implementation that no more batch statuses are gonna be emitted. It's a callback, not a method that actually prevents the emission (that I need to implemented myself). I don't know if it's clearer now

Reply:
Oh I see what you mean. It's a method that is called by the `download-manager` to prevent the statuses from being emitted, as opposed to something that the client can call to stop them 👍 makes sense 😄